invenio_oauthclient/contrib/cern.py

```diff
@@ -209,16 +211,68 @@ def account_setup(remote, token):
 
     response = remote.get(REMOTE_APP_RESOURCE_API_URL)
     user = token.remote_account.user
+    if not user:
+        raise Exception('`user` should not be None, something bad happened!')
 
     if response.status == requests.codes.ok:
         res = get_dict_from_response(response)
-        current_user.info['group'] = fetch_groups(res['Group'])
+
+        # update session object
+        groups_old = current_user.info['group']
+        groups_new = fetch_groups(res['Group'])
+        current_user.info['group'] = groups_new
```

Comment:
In legacy we did purge users, by regularly polling e-groups information: https://github.com/inveniosoftware/invenio/blob/legacy/modules/websession/lib/webgroup.py#L831

On the other hand empty e-groups were never deleted. Correct.

In fact if nowadays CERN e-groups offer an easy runtime API to fetch membership information, then there is no point to replicate e-group in Invenio.

P.s. other places where e-group were highly used were authorizations and in theory WebMessage :smile: